src/utils/sdfg/json_serializer.ts

```diff
@@ -23,10 +23,9 @@ function isDict(v: unknown): v is Record<string, unknown> {
         !(v instanceof Date);
 }
 
-function replacer(_name: string, val: unknown, orig_sdfg: unknown): unknown {
-    if (val && isDict(val) && val !== orig_sdfg && 'type' in val &&
-        val.type === 'SDFG') {
-        return JSON.stringify(val, (n, v) => replacer(n, v, val));
```

Comment:
What's the rationale with which we can drop this? What was it good for?

Reply:
It used to be a string, but now we keep it as a sub dictionary. So now we keep a backwards compatibility on reading but not writing.